core/pkg/launch/job_builder.go

```diff
@@ -679,7 +697,46 @@ func (j *JobBuilder) HandleUseArtifactRecord(record *service.Record) {
 		JobName:       strings.Split(useArtifact.Partial.JobName, ":")[0],
 		JobSourceInfo: jobSourceMetadata,
 	}
+}
 
+// Makes job input schema into a json string to be stored as artifact metdata.
+func (j *JobBuilder) makeJobMetadata(output *data_types.TypeRepresentation) (string, error) {
+	metadata := make(map[string]interface{})
+	if j.runConfig != nil {
+		include, exclude := j.getWandbConfigFilters()
+		runConfig := j.runConfig.FilterTree(include, exclude)
+		metadata[WandbConfigKey] = data_types.ResolveTypes(runConfig)
+	}
+	metadata = map[string]interface{}{"input_types": metadata}
+	if output != nil {
+		metadata["output_types"] = data_types.ResolveTypes(*output)
+	}
+	metadataBytes, err := json.Marshal(metadata)
+	if err != nil {
+		return "", err
+	}
+	return string(metadataBytes), nil
+}
+
+// Converts received LaunchWandbConfigParametersRecords into include and exclude paths.
+func (j *JobBuilder) getWandbConfigFilters() ([]runconfig.RunConfigPath, []runconfig.RunConfigPath) {
+	include := make([]runconfig.RunConfigPath, 0)
+	exclude := make([]runconfig.RunConfigPath, 0)
+	if len(j.wandbConfigParameters) > 0 {
+		for _, wandbConfigParameters := range j.wandbConfigParameters {
+			if wandbConfigParameters.IncludePaths != nil {
+				for _, includePath := range wandbConfigParameters.IncludePaths {
+					include = append(include, includePath.Path)
+				}
+			}
+			if wandbConfigParameters.ExcludePaths != nil {
+				for _, excludePath := range wandbConfigParameters.ExcludePaths {
+					exclude = append(exclude, excludePath.Path)
+				}
+			}
+		}
+	}
+	return include, exclude
```

Comment:
what happens if a path both included and excluded?

Reply:
oh so it won't be included based on the description i read somewhere else. right?

Reply:
yep, that was a stale docstring 😢